HSPI/Connector.cs

```diff
@@ -43,7 +43,7 @@ public static void Connect<TPlugin>(string[] args) where TPlugin : HspiBase, new
                     // let the plugin do it's thing, wait until it shuts down or the connection to HomeSeer fails.
                     try
                     {
-                        while (true)
+                        while (pollForShutdown)
```

Comment:
Also, based on the behavior you may want to rename it `exitOnShutdown` and say `if (exitOnShutdown) while true {...} }`. It’d make it easier to understand.

Reply:
Meh, I get what you're saying, but I don't know about `exitOnShutdown` either. I kinda half-ass threw `pollForShutdown` on the table just to see if it stuck. Apparently, it didn't. I'll think about it some more before merging, as changing the name later would result in a breaking change.